camel-core/src/main/java/org/apache/camel/component/mock/MockEndpoint.java

```diff
@@ -517,6 +517,9 @@ public void expectedMinimumMessageCount(int expectedCount) {
      * <b>Important:</b> This overrides any previous set value using {@link #expectedMessageCount(int)}
      */
     public void expectedHeaderReceived(final String name, final Object value) {
+        if (expectedCount == -1) {
+            expectedMessageCount(1);
```

Comment:
Shouldn't this be `setMinimumExpectedMessageCount(1);` ? I mean if expected count is not explicitly set, and you receive 5 exchanges all with the Header at the right value, the test should pass right? what do you think @davsclaus and @oscerd?